### PR TITLE
Added Navier-Stokes

### PR DESCRIPTION
**Pull request by arthbous**
Physical value and scaling are still left to do

### CODE REVIEW

**Comment by arthbous** [comment]
Almost done. The only values missing are the valencies of Na and Cl.

**Comment by maximilianmetti** [comment]
Ok. Those are both monovalent.  Na is +1e_c and Cl is -1e_c (e_c is the [elementary charge](https://en.wikipedia.org/wiki/Elementary_charge)).

**Comment by arthbous** [comment]
Chun divided by e so it's just +1 and -1.

**Comment by maximilianmetti** [comment]
dividing by e.... sound like Chun. ; ) So is this ready?

**Comment by arthbous** [comment]
Testing the refinement right now

**Comment by maximilianmetti** [comment]
Alright... I've made some updates to the refinement termination strategy, which I'm testing out.

**Comment by arthbous** [comment]
I don't think max_elements is working, it keeps refining even when I am above the limit set.

**Comment by maximilianmetti** [comment]
I think I had the same issue at some point... It will stop at some point, but I'm still looking into it.

**Comment by arthbous** [comment]
@maximilianmetti 
- All done. You can run the pnp_pb and pnp_ns_pb.  (One small issue the L2 error does not seem to decrease when the mesh increase)
- I also removed all the penalty in all the other pnp_ns tests.
- We need to merge it, that way I can use the new refinement for the sphere benchmark.

**Comment by maximilianmetti** [comment]
Is the error extremely small?  Could it just be round off?

**Comment by arthbous** [comment]
It is not small at all. and I am using the Error class.

**Comment by maximilianmetti** [comment]
So it doesn't converge to the right solution? Are the BCs and everything
correct?

-Maxx

**Comment by arthbous** [comment]
It does converges to the right solution

**Comment by arthbous** [comment]
My bad it does not...I am trying with different permittivity. There must be an error in the scaling..

**Comment by arthbous** [comment]
Maybe it's a problem with the boundary condition...

**Comment by arthbous** [comment]
@maximilianmetti 
All good now, It converges to the right solution and the L2 errors decrease with the mesh. We should merge it!

**Comment by maximilianmetti** [comment]
So this contains three experiments for the paper
 - PNP => PB
 - PNP+NS => PB + NS
 - NS with refinement

All these problems are steady, which I think demonstrates that our solver works for any time dependent problem 👍

**Comment by maximilianmetti** [comment]
Which benchmarks should I test?
I'm only trying out `pnp_pb`, `pnp_ns_pb`, `ns_spheres_refine`...

Is there a reason why we aren't doing refinement in `pnp_pb` and `pnp_ns_pb` to show that our `mesh_refinement` strategy works well too?  We have that in `pnp_refine_exact`, which I think completes the benchmark.

File: benchmarks/pnp_pb/main_ns.cpp [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+  std::vector<std::size_t> v1 = {0,1,2};
+  std::vector<double> v2 = {-Lx/2.0,-Ly/2.0,-Lz/2.0};
+  std::vector<double> v3 = { Lx/2.0,Ly/2.0,Lz/2.0};
+  auto BCdomain_xyz = std::make_shared<Dirichlet_Subdomain>(v1,v2,v3,1E-5);
+  dolfin::DirichletBC bc_sp1(pnp_ns_problem._functions_space[1],ub,sp_domain);
+  dolfin::DirichletBC bc_sp2(pnp_ns_problem._functions_space[1],ub,BCdomain_xyz);
+  bc_sp1.apply(*u_init.vector());
+  bc_sp2.apply(*u_init.vector());
+
+  printf("TOTO\n");fflush(stdout);
```
Comment (maximilianmetti) [inline benchmarks/pnp_pb/main_ns.cpp]: Can we remove this?

Comment (maximilianmetti) [reply benchmarks/pnp_pb/main_ns.cpp]: can we remove this?

Comment (arthbous) [reply benchmarks/pnp_pb/main_ns.cpp]: the print TOTO yes we can remove => done

**Comment by arthbous** [comment]
You can test pnp_pb, pnp_ns_pb, ns_spheres_refine and  ns_spheres

Yes, I can add the refinement if you insist. But for that we need to merge first, that way I test with the new refinement from develop.

**Comment by maximilianmetti** [comment]
We can either set the mesh size by hand several times, or we can use mesh refinement. I think it would be cool to show that our mesh refinement strategy is good enough to reduce error though.

**Comment by arthbous** [comment]
I agree. Let's merge this one and I will do new branch for the refinement of the two

**Comment by maximilianmetti** [comment]
I can merge this, but I've been running the `pnp_ns_pb` all night and it's still on the second iteration... It's stuck on the linear solver. How long does that step take for you?

**Comment by arthbous** [comment]
Your computer sucks, I pushed a coarser mesh try with that.

**Comment by maximilianmetti** [comment]
My computer sucks?? You can't even use your mouse!

**Pull request merged**